### DISCUSSION TITLE
DS3231

### DISCUSSION THREAD

**Discussion by gfn256** [arendst/Tasmota#17770] [Ideas]
Been using these DS3231 with some of my Tasmotas for some time, very useful for home-based NTP time server. Also with devices that need time immediately on boot up, like the clock display I built with a TM1637.
I was wondering whether we could access the inbuilt temperature sensor of this chip. I have extensively tested it in Arduino with the ds3231.h library, using the getTemp() function and it works very well. Accuracy is pretty usable in my environment.
It would be extremely useful for me to have Tasmota read this as a temp sensor.  This can't be too difficult to implement. Any ideas?

**Comment by barbudor** [comment]
That's sounds feasible probably based on an additional #define
however there doesn't seem to be a way to distinguish between DS1307 and DS3231, so it would have to be used only with DS3231

**Reply by barbudor** [reply]
I'm working on it and will test tonight

**Reply by gfn256** [reply]
Wonderful.  Will follow your progress. Thanks.

**Reply by barbudor** [reply]
Done https://github.com/arendst/Tasmota/pull/17778
Just wait for merge and you will be able to recompile
You need to add `#define DS3231_ENABLE_TEMP`

**Reply by gfn256** [reply]
Thanks so much. How can I know when its merged? Will this work with Tasmocompiler?
Once again thanks so much for such speedy results. TASMOTA ROCKS!

**Reply by barbudor** [reply]
You need to follow/subscribe the linked  PR
TasmoCompiler should work if you are using the development branch

**Reply by gfn256** [reply]
Thanks so much! Working Excellent.


![Screenshot 2023-01-24 171623](https://user-images.githubusercontent.com/93266249/214333024-ea75b80b-adec-43d1-8f08-7d472ee146df.png)

**Comment by CleberPeter** [comment]
Hi, first of all congratulations for the excellent work.

A doubt. in this situation the I2CDriver command should return address 0x68 as a member of the list?

**Comment by barbudor** [comment]
I2Cscan would return 0x68
I2cdriver is used to enable/disable the driver
The driver for DS3231 is number 26 so 'I2cdriver26 1/0'

https://tasmota.github.io/docs/I2CDEVICES/

**Reply by gfn256** [reply]
Note that depending on the RTC module you have, I2Cscan may find a second I2C device which is the EEPROM on the module.

**Reply by CleberPeter** [reply]
Perfect. Thanks for the prompt support, tested on ESP32 and it's working properly.
